### PR TITLE
Avoid potential crash on signal disconnection

### PR DESCRIPTION
**Pull request by RandomShaper**
I have not been able to come up with an MRP. It seems that on very complex resource dependency graphs, with signal connections among them, there's the possibility of hitting a pathological case where a signal connection is no longer in the map, which causes as segfault.

### CODE REVIEW

**Comment by AThousandShips** [comment]
Seems you forgot to include some files

**Comment by RandomShaper** [comment]
You're right. Fixed.

File: core/object/object.cpp [hunk 1]
```diff
@@ -2097,10 +2097,10 @@ Object::~Object() {
...
 	while (connections.size()) {
 		Connection c = connections.front()->get();
-		bool disconnected = c.signal.get_object()->_disconnect(c.signal.get_name(), c.callable, true);
-		if (unlikely(!disconnected)) {
-			// If the disconnect has failed, abandon the connection to avoid getting trapped in an infinite loop here.
-			connections.pop_front();
+		Object *obj = c.callable.get_object();
+		bool disconnected = false;
+		if (likely(obj)) {
+			disconnected = c.signal.get_object()->_disconnect(c.signal.get_name(), c.callable, true);
```
Comment (akien-mga) [inline core/object/object.cpp:2103]: You're not doing anything with `disconnected`, so this raises a warning.

Comment (RandomShaper) [reply core/object/object.cpp:2103]: Oh, right. It's no surprise I've kept messing up this PR as my last work hour today has been a waterfall of mistakes. 😅 I'll fix it and push.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.3.1.

**Pull request merged**